hoover/search/es.py

```diff
@@ -103,36 +149,42 @@ def extract_info(ex):
 
 
 def delete_index(collection_id, ok_missing=False):
-    es.indices.delete(
-        index=_index_name(collection_id),
-        ignore=[404] if ok_missing else [],
-    )
+    with elasticsearch() as es:
+        es.indices.delete(
+            index=_index_name(collection_id),
+            ignore=[404] if ok_missing else [],
+        )
 
 
 def delete_all():
-    for index in es.indices.get(index='*'):
-        if index.startswith(settings.ELASTICSEARCH_INDEX_PREFIX):
-            es.indices.delete(index=index)
+    with elasticsearch() as es:
+        for index in es.indices.get(index='*'):
+            if index.startswith(settings.ELASTICSEARCH_INDEX_PREFIX):
+                es.indices.delete(index=index)
 
 
 def refresh():
-    es.indices.refresh()
+    with elasticsearch() as es:
+        es.indices.refresh()
 
 
 def count(collection_id):
+    es = Elasticsearch(settings.HOOVER_ELASTICSEARCH_URL)
```

Comment:
It's catched anyway and a `SearchError` is raised. Using the contextmanager in that way, I would have to check for the reason to be a specific string, and that will be a bit magic.

The fix is below: instead of using the `NotFoundError` for logic, the index is checked for with `es.indices.exists` before creating an alias for it.